account_payment_purchase/models/account_invoice.py

```diff
@@ -10,8 +10,16 @@ class AccountMove(models.Model):
 
     @api.onchange("purchase_vendor_bill_id", "purchase_id")
     def _onchange_purchase_auto_complete(self):
-        new_mode = self.purchase_id.payment_mode_id.id or False
-        new_bank = self.purchase_id.supplier_partner_bank_id.id or False
+
+        new_mode = (
+            self.purchase_vendor_bill_id.purchase_order_id.payment_mode_id.id
+            or self.purchase_id.payment_mode_id.id
+        )
+        new_bank = (
+            self.purchase_vendor_bill_id.purchase_order_id.supplier_partner_bank_id.id
+            or self.purchase_id.supplier_partner_bank_id.id
+        )
```

Comment:
Little question, shoudn't this have into account that `purchase_vendor_bill_id` can also have a relation to an invoice?
So it would look something like this:
```suggestion
        new_mode = (
            self.purchase_vendor_bill_id.purchase_order_id.payment_mode_id.id
            or self.purchase_vendor_bill_id.vendor_bill_id.payment_mode_id.id
            or self.purchase_id.payment_mode_id.id
        )
        new_bank = (
            self.purchase_vendor_bill_id.purchase_order_id.supplier_partner_bank_id.id
            or self.purchase_vendor_bill_id.vendor_bill_id.partner_bank_id.id
            or self.purchase_id.supplier_partner_bank_id.id
        )
```

Reply:
Or maybe, add this on top of the function :thinking: :
```
if self.purchase_vendor_bill_id.vendor_bill_id:
    return super()._onchange_purchase_auto_complete()
```